server/datastore/mysql/hosts.go

```diff
@@ -1713,3 +1713,32 @@ ON DUPLICATE KEY UPDATE
 
 	return nil
 }
+
+func (ds *Datastore) HostIDsByPlatform(
+	ctx context.Context,
+	platform string,
+	osVersion string,
+) ([]uint, error) {
+	var ids []uint
+	var filters []goqu.Expression
+
+	stmt := dialect.From("hosts").Select("id")
+	if platform != "" {
```

Comment:
Fleet's convention so far is to use `platform *string` as argument for these cases.